packages/bot-config-utils/src/bot-config-utils.ts

```diff
@@ -174,60 +175,70 @@ export class ConfigChecker<ConfigType> {
     commitSha: string,
     prNumber: number
   ): Promise<void> {
-    const listFilesParams = {
-      owner: owner,
-      repo: repo,
-      pull_number: prNumber,
-      per_page: 100,
-    };
-    let errorText = '';
-    const files = await octokit.paginate(
-      octokit.pulls.listFiles,
-      listFilesParams
-    );
-    for (const file of files) {
-      if (file.status === 'removed') {
-        continue;
-      }
-      logger.debug(`file: ${file.filename}`);
-      if (this.badConfigPaths.indexOf(file.filename) > -1) {
-        // Trying to add a config file with a wrong file extension.
-        errorText +=
-          `You tried to add ${file.filename}, ` +
-          `but the config file must be ${this.configPath}\n`;
-      }
-      if (file.filename === this.configPath) {
-        const blob = await octokit.git.getBlob({
-          owner: owner,
-          repo: repo,
-          file_sha: file.sha,
-        });
-        const fileContents = Buffer.from(blob.data.content, 'base64').toString(
-          'utf8'
-        );
-        const result = validateConfig<ConfigType>(fileContents, this.schema, {
-          additionalSchemas: this.additionalSchemas,
-        });
-        if (result.isValid) {
-          this.config = result.config as ConfigType;
-        } else {
-          errorText += result.errorText;
+    // Sometimes the head branch is gone.
+    // In that case, the requests for fetching files might fail with 404.
+    // We can just ignore those cases.
+    try {
+      const listFilesParams = {
+        owner: owner,
+        repo: repo,
+        pull_number: prNumber,
+        per_page: 100,
+      };
+      let errorText = '';
+      const files = await octokit.paginate(
+        octokit.pulls.listFiles,
+        listFilesParams
+      );
+      for (const file of files) {
+        if (file.status === 'removed') {
```

Comment:
Should we log that we attempted to grab the file or somehow bubble up an alert? Won't it just fail silently down the line (since there's no config)? I might not be understanding the full context though.

Reply:
This code is iterating all the files in the PR:
- If it's removed, we just skip it
- If the filename is the bot's config file, we want to check the schema
- Also the filename looks to have a wrong file extension, we report that too

I don't think we need to do anything for 'removed' case.